src/core/storage/index.js

```diff
@@ -0,0 +1,46 @@
+// @flow
+// $FlowIgnore[missing-export]
+import {TextEncoder, TextDecoder} from "util";
+
+/**
+ * A byte string represented as a Uint8Array. Each item in a byteString array
+ * is a Uint8 representing a byte value.
+ */
+export type ByteString = Uint8Array;
+
+/**
+ * Data Storage is meant as a low-level interface to data that is not
+ * already in process memory. If the value for the key entered cannot be found,
+ * an error should throw.
```

Comment:
documentation:

  - Again, “Data Storage” isn’t anything; `DataStorage` is the name of
    this interface.
  - Don’t tell me that it’s meant as something. You are the author and
    may be authoritative. If it’s meant as a low-level interface to
    data, say “A low-level interface to data […]”.
  - There’s no _requirement_ that the data not be in process memory: a
    `MemoryDataStorage` is in fact a useful testing utility that we may
    want to define. Rephrase to make it clear that implementations are
    merely *permitted* to perform I/O?
  - Errors do not throw; errors are thrown.
  - “the value for the key”—what key? `DataStorage` does not have keys.
    This comment is appropriate for the `get` method, and should live
    there instead.